tiledb/sm/query/query.cc

```diff
@@ -1526,7 +1543,8 @@ Status Query::submit() {
           "Error in query submission; remote array with no rest client."));
 
     if (status_ == QueryStatus::UNINITIALIZED) {
-      RETURN_NOT_OK(create_strategy());
+      // Create query strategy, skip checks for serialization.
```

Comment:
What issue hits here when you pass in false?

Reply:
Looks like WriterBase was throwing an exception for this check, since it isn't aware of dimension label buffers. This is only thrown for write queries using dim labels exclusively.

https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/query/writers/writer_base.cc#L117

I hit this error client side and seen the `skip_checks_serialization` param, it seemed like passing true here made sense for remote queries. Happy to take another look if this isn't correct. Here is one of the tests that fail on the `Without array data` section if I pass in false here, as an example.

https://github.com/TileDB-Inc/TileDB/blob/dev/test/src/test-cppapi-dense-array-dimension-label.cc#L413

Thanks for pointing this out, as-is I found a change that should have been reverted after I looked into [Julia's comment](https://github.com/TileDB-Inc/TileDB/pull/4084#discussion_r1191531504) above.

Reply:
I don't see the failure locally... I changed the call to false to see what happens in CI.

Reply:
The test only fails pointing at REST so I don't think CI will hit this specific failure, we can revert it though if it doesn't look right. I just setup configs to run all of the dimension label tests through a local REST server and this was one of the failures I hit.